pcli/src/command/stake.rs

```diff
@@ -0,0 +1,145 @@
+use anyhow::{anyhow, Result};
+use penumbra_crypto::Value;
+use penumbra_proto::thin_wallet::{thin_wallet_client::ThinWalletClient, ValidatorRateRequest};
+use penumbra_stake::{Epoch, IdentityKey, RateData, STAKING_TOKEN_ASSET_ID};
+use penumbra_wallet::ClientState;
+use structopt::StructOpt;
+
+use crate::Opt;
+
+#[derive(Debug, StructOpt)]
+pub enum StakeCmd {
+    /// Deposit stake into a validator's delegation pool.
+    Delegate {
+        /// The identity key of the validator to delegate to.
+        #[structopt(long)]
+        to: String,
+        /// The amount of stake to delegate.
+        amount: String,
+        /// The transaction fee (paid in upenumbra).
+        #[structopt(long, default_value = "0")]
+        fee: u64,
+        /// Optional. Only spend funds originally received by the given address index.
+        #[structopt(long)]
+        source: Option<u64>,
+    },
+    /// Withdraw stake from a validator's delegation pool.
+    Undelegate {
+        /// The identity key of the validator to withdraw delegation from.
+        #[structopt(long)]
+        from: String,
+        /// The amount of stake to delegate.
+        amount: String,
+        /// The transaction fee (paid in upenumbra).
+        #[structopt(long, default_value = "0")]
+        fee: u64,
+        /// Optional. Only spend funds originally received by the given address index.
+        #[structopt(long)]
+        source: Option<u64>,
+    },
+    /// Redelegate stake from one validator's delegation pool to another.
+    Redelegate {
+        /// The identity key of the validator to withdraw delegation from.
+        #[structopt(long)]
+        from: String,
+        /// The identity key of the validator to delegate to.
+        #[structopt(long)]
+        to: String,
+        /// The amount of stake to delegate.
+        amount: String,
+        /// The transaction fee (paid in upenumbra).
+        #[structopt(long, default_value = "0")]
+        fee: u64,
+        /// Optional. Only spend funds originally received by the given address index.
+        #[structopt(long)]
+        source: Option<u64>,
+    },
+    /// Display this wallet's delegations and their value.
+    Show,
```

Comment:
@avahowell and I thought it made sense to have this be separate from the `balance` command; the `balance` command will show the amounts of all of the delegation tokens, but doesn't know that they're anything other than normal tokens.

It'd be useful to have staking-specific information, like showing the equivalent amount in unbonded stake, bonded stake, showing more details on validators (like the human-readable name), etc., and putting all that into the normal `balance` command seems a bit messy.